sdk/eventhub/Azure.Messaging.EventHubs/src/EventHubConsumerClientOptions.cs

```diff
@@ -128,6 +126,36 @@ public int PrefetchCount
         ///
         public bool TrackLastEnqueuedEventInformation { get; set; } = true;
 
+        /// <summary>
+        ///   Gets or sets the options used for configuring the connection to the Event Hubs service.
+        /// </summary>
+        ///
+        public EventHubConnectionOptions ConnectionOptions
+        {
+            get => _connectionOptions;
+            set
+            {
+                Argument.AssertNotNull(value, nameof(ConnectionOptions));
```

Comment:
I know it won't make any difference in the end, but why preventing the user from setting the `ConnectionOptions` to null here and in the other clients? The `EventHubConnection` constructors allow it.

Reply:
In a nutshell, to make it easier to reason about things.   We default it at construction, which makes it explicit and clear for callers to understand.  Your argument makes sense, though.   I'm on the fence.  

I'll sleep on it today and weigh in with an update tomorrow.

Reply:
I'm going to leave this as-is.   At this point, a developer has chosen to take ownership of the options, which they could have left to the automatic defaults.  By so doing, they've made the decision to provide the configuration needed for a client.  The connection and retry options are automatically populated with the defaults and if a developer wishes to revert to default values after changing things, for some reason, this can be done easily be setting them to a `new` copy.

I can't come up with a single compelling case of why someone would legitimately need or want to explicitly set them to `null`.  While it technically would be fine, as there would be a default automatically created, I feel that it would allow putting a set of `EventHubXXXClientOptions` into a state that is isn't really valid.  Connection options are always needed, its just a matter of who provides them and, in this case, a developer has chosen to own the options and should ensure all relevant ones are provided.

If you've got some scenarios in mind that I'm overlooking, I'm happy to reconsider.